include/mp/proxy.h

```diff
@@ -7,35 +7,31 @@
 
 #include <mp/util.h>
 
-#include <array>
 #include <cassert>
 #include <functional>
 #include <list>
+#include <memory>
 #include <stddef.h>
 #include <tuple>
 #include <type_traits>
 #include <utility>
+#include <variant>
```

Comment:
IWYU's suggestion here appears to be incorrect.

Reply:
> IWYU's suggestion here appears to be incorrect.

I agree with @maflcko. I didn't mean the current code should be changed. Just pointed at potential IWYU bug.

Reply:
re: https://github.com/bitcoin-core/libmultiprocess/pull/184#discussion_r2166968873

Makes sense, added keep pragma.